src/test/java/org/junitpioneer/jupiter/params/DisableIfParameterExtensionTests.java

```diff
@@ -0,0 +1,149 @@
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.junitpioneer.testkit.ExecutionResults;
+import org.junitpioneer.testkit.PioneerTestKit;
+
+/**
+ * Oscar Wilde: Requiescat is in the public domain
+ */
+@DisplayName("DisableIfParameterExtension")
+class DisableIfParameterExtensionTests {
+
+	@Nested
+	@DisplayName("when configured correctly")
+	class CorrectConfigurationTests {
+
+		@Test
+		@DisplayName("disables tests when parameter contains any value from the 'contains' array")
+		void interceptContains() {
+			ExecutionResults results = PioneerTestKit
+					.executeTestMethodWithParameterTypes(CorrectConfigTestCases.class, "interceptContains",
+						String.class);
+
+			assertThat(results).hasNumberOfSucceededTests(2).hasNumberOfAbortedTests(2);
+		}
+
+		@Test
+		@DisplayName("disables tests when any parameter contains any value from the 'contains' array")
+		void interceptContainsAny() {
+			ExecutionResults results = PioneerTestKit
+					.executeTestMethodWithParameterTypes(CorrectConfigTestCases.class, "interceptContainsAny",
+						String.class, String.class);
+
+			assertThat(results).hasNumberOfAbortedTests(2);
+		}
+
+		@Test
+		@DisplayName("disables tests when parameter matches any regex from the 'matches' array")
+		void interceptMatches() {
+			ExecutionResults results = PioneerTestKit
+					.executeTestMethodWithParameterTypes(CorrectConfigTestCases.class, "interceptMatches",
+						String.class);
+
+			assertThat(results).hasNumberOfSucceededTests(2).hasNumberOfAbortedTests(2);
+		}
+
+		@Test
+		@DisplayName("disables tests when any parameter matches any regex from the 'matches' array")
+		void interceptMatchesAny() {
+			ExecutionResults results = PioneerTestKit
+					.executeTestMethodWithParameterTypes(CorrectConfigTestCases.class, "interceptMatchesAny",
+						String.class, String.class);
+
+			assertThat(results).hasNumberOfAbortedTests(2);
+		}
+
+		@Test
+		@DisplayName("disables tests if parameter matches regex from 'matches' or contains value from 'contains'")
+		void interceptMatchesAndContains() {
+			ExecutionResults results = PioneerTestKit
+					.executeTestMethodWithParameterTypes(CorrectConfigTestCases.class, "interceptBoth", String.class);
+
+			assertThat(results).hasNumberOfSucceededTests(1).hasNumberOfAbortedTests(3);
+		}
+
+	}
+
+	@Nested
+	@DisplayName("when not configured correctly")
+	class MisconfigurationTests {
+
+		@Test
+		@DisplayName("throws an exception if both 'matches' and 'contains' is missing")
+		void missingValues() {
+			ExecutionResults results = PioneerTestKit
+					.executeTestMethodWithParameterTypes(BadConfigTestCases.class, "missingValues", String.class);
+
+			assertThat(results).hasNumberOfFailedTests(3);
+		}
+
+	}
```

Comment:
I'm confused why
1) you don't check for the particular exceptions you throw
2) Why sonar claims full coverage (speaking of the part with the three different exceptions)

🤷‍♂️

Reply:
Do you mean the `TestAbortedException`?

Reply:
I mean you have three checks about misconfiguration you do in the extension, but only one test case and I don't get why SONAR does not complain 🤷‍♂️

Reply:
It's line coverage, not part coverage. If the line is called, it's considered covered. Example:
```java
/* 100% coverage */
Optional.of(new Object())
.orElseThrow(() -> new RuntimeException("Never thrown!"));

/* < 100% coverage */
Optional.of(new Object())
.orElseThrow(
    () -> new RuntimeException("Never thrown!")
);
```